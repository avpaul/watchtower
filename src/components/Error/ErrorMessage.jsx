import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({ message, customStyle }) => {
  const defaultStyle = {
    width: '100%',
    margin: '1rem auto',
    fontSize: '2rem',
    textAlign: 'center',
  };

  const errorMessageStyle = { ...defaultStyle, ...customStyle };
  return (
    <div style={errorMessageStyle}>{message}</div>
  );
};

ErrorMessage.defaultProps = {
  customStyle: {},
};

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  customStyle: PropTypes.objectOf(PropTypes.string),
};

export default ErrorMessage;
