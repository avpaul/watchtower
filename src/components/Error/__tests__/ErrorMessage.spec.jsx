import React from 'react';
import { shallow } from 'enzyme';
import ErrorMessage from '../ErrorMessage';

it('should match snapshot', () => {
  const wrapper = shallow(<ErrorMessage message="test" />);
  expect(wrapper).toMatchSnapshot();
});
