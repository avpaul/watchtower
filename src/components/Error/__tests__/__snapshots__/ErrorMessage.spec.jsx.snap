// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  style={
    Object {
      "fontSize": "2rem",
      "margin": "1rem auto",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  test
</div>
`;
